source/module_rdmft/rdmft.h

```diff
@@ -0,0 +1,254 @@
+    TwoCenterBundle* two_center_bundle = nullptr;
+    // Local_Orbital_Charge* loc = nullptr;  // would be delete in the future
+    /****** these parameters are passed in from outside, don't need delete ******/
+
+    int nk_total = 0;
+    int nbands_total;
+    int nspin = 1;
+    std::string XC_func_rdmft;
+    double alpha_power = 0.656; // 0.656 for soilds, 0.525 for dissociation of H2, 0.55~0.58 for HEG
+
+    // natrual occupation numbers and wavefunction
+    ModuleBase::matrix occ_number;
+    psi::Psi<TK> wfc;
+    ModuleBase::matrix wg;
+    ModuleBase::matrix wk_fun_occNum;
+
+    // store the gradients of Etotal with respect to the natural occupation numbers and wfc respectively
+    ModuleBase::matrix occNum_wfcHamiltWfc;
+    psi::Psi<TK> occNum_HamiltWfc;
+
+    // E_RDMFT[4] stores ETV, Ehartree, Exc, Etotal respectively
+    double E_RDMFT[4] = {0.0};
+    // std::vector<double> E_RDMFT(4);
+
+    hamilt::HContainer<TR>* HR_TV = nullptr;
+    hamilt::HContainer<TR>* HR_hartree = nullptr;
+    // hamilt::HContainer<TR>* HR_XC = nullptr;
+    hamilt::HContainer<TR>* HR_dft_XC = nullptr;
+    hamilt::HContainer<TR>* HR_exx_XC = nullptr;
+    // hamilt::HContainer<TR>* HR_local = nullptr;
+
+    std::vector<TK> HK_TV;
+    std::vector<TK> HK_hartree;
+    std::vector<TK> HK_exx_XC;
+    std::vector<TK> HK_dft_XC;
+
+    hamilt::HS_Matrix_K<TK>* hsk_TV;
+    hamilt::HS_Matrix_K<TK>* hsk_hartree;
+    hamilt::HS_Matrix_K<TK>* hsk_dft_XC;
+    hamilt::HS_Matrix_K<TK>* hsk_exx_XC;
+
+    std::vector<TK> HK_XC;
+    std::vector<TK> HK_local;
+    std::vector< std::vector<TK> > DM_XC_pass;
+    // ModuleDirectMin::ProdStiefelVariable HK_RDMFT_pass;
+    // ModuleDirectMin::ProdStiefelVariable HK_XC_pass;
+
+    ModuleBase::matrix Etotal_n_k;
+    ModuleBase::matrix wfcHwfc_TV;
+    ModuleBase::matrix wfcHwfc_hartree;
+    ModuleBase::matrix wfcHwfc_XC;
+    ModuleBase::matrix wfcHwfc_exx_XC;
+    ModuleBase::matrix wfcHwfc_dft_XC;
+
+    psi::Psi<TK> H_wfc_TV;
+    psi::Psi<TK> H_wfc_hartree;
+    psi::Psi<TK> H_wfc_XC;
+    psi::Psi<TK> H_wfc_exx_XC;
+    psi::Psi<TK> H_wfc_dft_XC;
+
+    // just for temperate. in the future when realize psiDotPsi() without pzgemm_/pdgemm_,we don't need it
+    std::vector<TK> Eij_TV;
+    std::vector<TK> Eij_hartree;
+    std::vector<TK> Eij_XC;
+    std::vector<TK> Eij_exx_XC;
+
+    hamilt::OperatorLCAO<TK, TR>* V_ekinetic_potential = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_nonlocal = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_local = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_hartree = nullptr;
+    // hamilt::OperatorLCAO<TK, TR>* V_XC = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_exx_XC = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_dft_XC = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_hartree_XC = nullptr;
+    // bool get_V_local_temp = true;
+
+    Exx_LRI<double>* Vxc_fromRI_d = nullptr;
+    Exx_LRI<std::complex<double>>* Vxc_fromRI_c = nullptr;
```

Comment:
Oops, you need to add new `Exx_LRI` objects here...
Seems they share the same `ccp_type` and `ccp_rmesh_times` as those in esolver_ks_lcao.h, namely `exx_lri_double` and `exx_lri_complex`. The only difference is its `Hexxs` member calculted with different density matrices by `cal_exx_elec`, right? But the new objects here double the memory cost of ABFs, two-center tables and RI- C, V tensors...
I think it's not a good design of classs `Exx_LRI`.

Reply:
yes,  the only difference is its Hexxs member calculted with different density matrices by cal_exx_elec.

and I give ccp_rmesh_times as shown below (ccp_type is similar)
![ccp_rmesh_times_forRDMFT](https://github.com/user-attachments/assets/cefa9a6a-c913-42e9-ab55-772762347eff)